ext/sysvmsg/sysvmsg.c

```diff
@@ -371,11 +371,15 @@ PHP_FUNCTION(msg_send)
 		php_var_serialize(&msg_var, message, &var_hash);
 		PHP_VAR_SERIALIZE_DESTROY(var_hash);
 
+		message_len = smart_str_get_len(&msg_var);
+		if (!message_len) {
+			RETURN_FALSE;
+		}
+
 		/* NB: php_msgbuf is 1 char bigger than a long, so there is no need to
 		 * allocate the extra byte. */
-		messagebuffer = safe_emalloc(ZSTR_LEN(msg_var.s), 1, sizeof(struct php_msgbuf));
-		memcpy(messagebuffer->mtext, ZSTR_VAL(msg_var.s), ZSTR_LEN(msg_var.s) + 1);
-		message_len = ZSTR_LEN(msg_var.s);
+		messagebuffer = safe_emalloc(message_len, 1, sizeof(struct php_msgbuf));
+		memcpy(messagebuffer->mtext, ZSTR_VAL(msg_var.s), message_len + 1);
 		smart_str_free(&msg_var);
```

Comment:
Did not test it yet ; will do in the next hour, but that should work too, I just wanted to have an early exit/not bothering creating an empty message.

Reply:
> I just wanted to have an early exit/not bothering creating an empty message.

Makes sense, but still could use proper `smart_str` APIs instead of accessing the underlying `zend_string` directly.